jbang/tnb.java

```diff
@@ -0,0 +1,677 @@
+                        terminal.writer().println("\nStopping deployment of " + svcClazz.getSimpleName());
+                        return null;
+                    }
+                    FieldUtils.writeField(service, "account", account, true);
+                }
+            }
+            service.beforeAll(null);
+            services.add(service);
+
+            System.setProperty(OpenshiftConfiguration.USE_OPENSHIFT, previousUseOCP);
+
+            return service;
+        }
+
+        public static List<String> findAllRegisteredServices() {
+            List<String> serviceNames = new ArrayList<>();
+            try {
+                final List<URL> tnbServices = Collections.list(Thread.currentThread().getContextClassLoader().getResources("META-INF/services/"))
+                    .stream()
+                    .filter(url -> url.getPath().contains("tnb"))
+                    .collect(Collectors.toList());
+
+                for (URL tnbDependency : tnbServices) {
+                    final String jarPath = tnbDependency.getPath().replace("file:", "").replace("!/META-INF/services/", "");
+                    try (ZipFile zip = new ZipFile(jarPath)) {
+                        serviceNames.addAll(Collections.list(zip.entries()).stream().map(ZipEntry::getName)
+                            .filter(name -> name.startsWith("META-INF/services/software.tnb")).map(s -> s.replace("META-INF/services/", "")).collect(
+                                Collectors.toList()));
+                    } catch (IOException e) {
+                        System.err.println("Failed to load System-X service: " + e);
+                    }
+                }
+            } catch (IOException e) {
+                return Collections.emptyList();
+            }
+            return serviceNames;
+        }
+
+        public static Set<Class<? extends Service<?, ?, ?>>> getRegisteredServices() {
+            return registeredServices;
+        }
+
+        @SuppressWarnings("unchecked")
+        public static Set<Class<? extends Service<?, ?, ?>>> initSvcClasses() {
+            return findAllRegisteredServices().stream()
+                .map(svc -> Try.call(() -> (Class<Service<?, ?, ?>>) Class.forName(svc))
+                    .ifFailure(ex -> terminal.writer().println("Failure while loading class: " + ex.getMessage()))
+                    .toOptional().orElse(null)
+                )
+                .collect(Collectors.toSet());
+        }
+
+        public static void removeService(Service<?, ?, ?> service) {
+            services.remove(service);
+            try {
+                service.afterAll(null);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        public static List<Service<?, ?, ?>> getServices() {
+            return services;
+        }
+    }
+
+    public static class DumpHelper {
+
+        public static Map<String, Object> dumpObject(Object obj) {
```

Comment:
this could use some comments what it does (I guess it will `dump` the `object` based on its name, but a bit of detail would help)